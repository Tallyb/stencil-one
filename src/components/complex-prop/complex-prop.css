.nice {
  background: yellow;
  color: purple;

}