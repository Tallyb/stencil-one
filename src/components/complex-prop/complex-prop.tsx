import { Component, Prop, h, JSX} from '@stencil/core';

@Component({
  tag: 'my-complex-prop',
  styleUrl: 'complex-prop.css',
  shadow: true
})
export class MyComplexPropComponent {

  @Prop({mutable: true}) values: Array<string> = [];

  toUpper (items: Array<string>) {
    return items.map( i => i.toUpperCase());
  }
  componentWillLoad() {
    this.values = this.toUpper(this.values);
    console.log('Will load', this.values);
  }

  render() : JSX.Element {
    return (
      <div class="nice">
      {this.values.map((item) => {
            return <div class="item">
                <span>{item}</span>
            </div>
        })}
      </div>
    );
  }
}
