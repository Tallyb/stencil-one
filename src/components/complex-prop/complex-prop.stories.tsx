import { MyComplexPropComponent } from './complex-prop';
import {h} from '@stencil/core';

export default {
  title: 'My Complex',
  components: MyComplexPropComponent
};

export const Default = () => 
  (<my-complex-prop values={['aaa', 'bbb']}></my-complex-prop>);

