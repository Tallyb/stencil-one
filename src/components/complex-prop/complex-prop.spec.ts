import { newSpecPage } from '@stencil/core/testing';

import { MyComplexPropComponent } from './complex-prop';

describe('complex prop', () => {

    it ('should change to upper case', () => {
        let cmp = new MyComplexPropComponent();
        let res = cmp.toUpper(['aaa', 'bbb', 'ccc']);
        expect(res).toEqual(['AAA', 'BBB', 'CCC']);
    });

    it('should render', async () => {
        const page = await newSpecPage({
            components: [MyComplexPropComponent],
            html: `
                <my-complex-prop>
                </my-complex-prop>
            `
        });
        page.rootInstance.values = ['aaa', 'bbb', 'ccc'];
        await page.waitForChanges();
        expect(page.root).toEqualHtml(`
         <my-complex-prop>
           <mock:shadow-root>
             <div class="nice">
               <div class="item">
                 <span>
                   aaa
                 </span>
               </div>
               <div class="item">
                 <span>
                   bbb
                 </span>
               </div>
               <div class="item">
                 <span>
                   ccc
                 </span>
               </div>
             </div>
           </mock:shadow-root>
         </my-complex-prop>
        `)
    });
    it('should render with data', async () => {
        const page = await newSpecPage({
            components: [MyComplexPropComponent],
            html: '<div></div>'
        });
        let cmp = page.doc.createElement('my-complex-prop');
        (cmp as any).values = ['aaa', 'bbb', 'ccc'];
        page.root.appendChild(cmp);
        await page.waitForChanges();
        expect(page.root).toEqualHtml(`
        <div>
          <my-complex-prop>
            <mock:shadow-root>
              <div class=\"nice\">
                <div class=\"item\">
                  <span>
                    AAA
                  </span>
                </div>
                <div class=\"item\">
                  <span>
                    BBB
                  </span>
                </div>
                <div class=\"item\">
                  <span>
                    CCC
                  </span>
                </div>
              </div>
            </mock:shadow-root>
          </my-complex-prop>
        </div>
        `)
    });


});