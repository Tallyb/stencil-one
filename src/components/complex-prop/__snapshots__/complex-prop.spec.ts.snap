// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex prop should render 1`] = `
<my-complex-prop>
  <mock:shadow-root>
    <div class="nice">
      <div class="item">
        <span>
          aaa
        </span>
      </div>
      <div class="item">
        <span>
          bbb
        </span>
      </div>
      <div class="item">
        <span>
          ccc
        </span>
      </div>
    </div>
  </mock:shadow-root>
</my-complex-prop>
`;

exports[`complex prop should render with data 1`] = `
<my-complex-prop>
  <mock:shadow-root>
    <div class="nice">
      <div class="item">
        <span>
          AAA
        </span>
      </div>
      <div class="item">
        <span>
          BBB
        </span>
      </div>
      <div class="item">
        <span>
          CCC
        </span>
      </div>
    </div>
  </mock:shadow-root>
</my-complex-prop>
`;
